### PR TITLE
Fix security by upgrading to Gulp 4.0

### PR DESCRIPTION
**Pull request by WardenGnaw**
Gulp 4.0 introduces a breaking change. This fixes the vulnerabilities but test code will need to be refactored to properly spawn the VsCode test.

### CODE REVIEW

**Review by sean-mcmanus — APPROVED** [review]

File: .travis.yml [hunk 1]
```diff
@@ -35,7 +35,9 @@ script:
   - npm run tslint
   # pr-check needs to run before test. test modifies package.json.
   - npm run pr-check
-  - npm run test
+  - npm run unitTests
+  # TODO: Merge tests into one group... Silly Gulp 4.0
```
Comment (bobbrow) [inline .travis.yml:39]: consider removing "silly" from the comment. 😄

**Review by bobbrow — APPROVED** [review]

File: .travis.yml (hunk 1 above)
Comment (sean-mcmanus) [reply .travis.yml:39]: Really? I liked that comment :)

**Comment by sean-mcmanus** [comment]
Can this get checked in?

**Comment by WardenGnaw** [comment]
I haven't fully tested it locally on how it affects dev work. I just made it to work with Travis-CI.

**Comment by WardenGnaw** [comment]
Updating to a newer PR.

**Pull request closed without merging**